site/docs/flink.md

```diff
@@ -0,0 +1,277 @@
+);
+```
+
+We could execute the sql command `use catalog hive_catalog` to use one of the catalog.
+
+## DDL commands
+
+### `CREATE DATABASE`
+
+By default, iceberg will use the `default` database in flink. Using the following example to create a separate database:
+
+```sql
+CREATE DATABASE iceberg_db;
+USE iceberg_db;
+```
+
+### `CREATE TABLE`
+
+```sql
+CREATE TABLE hive_catalog.default.sample (
+    id BIGINT COMMENT 'unique id',
+    data STRING
+);
+```
+
+Table create commands support most common [flink create clauses](https://ci.apache.org/projects/flink/flink-docs-release-1.11/dev/table/sql/create.html#create-table) now, including: 
+
+* `PARTITION BY (column1, column2, ...)` to configure partitioning.
+* `COMMENT 'table document'` to set a table description.
+* `WITH ('key'='value', ...)` to set [table configuration](../configuration)
+
+Currently, it does not support computed column, primary key and watermark definition etc.
+
+### `PARTITIONED BY`
+
+To create a partition table, use `PARTITIONED BY`:
+
+```sql
+CREATE TABLE hive_catalog.default.sample (
+    id BIGINT COMMENT 'unique id',
+    data STRING
+) PARTITIONED BY (data);
+```
+
+Apache Iceberg support hidden partition but apache flink don't support partitioning by a function on columns, so we've no way to support hidden partition in flink DDL now, we will improve apache flink DDL in future.
+
+### `ALTER TABLE`
+
+Iceberg only support altering table properties in flink 1.11 now.
+
+```sql
+ALTER TABLE hive_catalog.default.sample SET ('write.format.default'='avro')
+```
+
+### `ALTER TABLE .. RENAME TO`
+
+```sql
+ALTER TABLE hive_catalog.default.sample RENAME TO hive_catalog.default.new_sample;
+```
+
+### Adding/Removing/Renaming table columns
+
+Iceberg does not support adding/removing/renaming table columns now.
+
+### Altering tables' partition keys.
+
+Iceberg does not support altering partition keys now.
+
+### `DROP TABLE`
+
+To delete a table, run:
+
+```sql
+DROP TABLE hive_catalog.default.sample;
+```
+
+## Querying with SQL
+
+Iceberg does not support streaming read or batch read in flink now, it's still working in-progress.
+
+## Writing with SQL
+
+Iceberg support both `INSERT INTO` and `INSERT OVERWRITE` in flink 1.11 now.
+
+### `INSERT INTO`
+
+To append new data to a table with a flink streaming job, use `INSERT INTO`:
+
+```sql
+INSERT INTO hive_catalog.default.sample VALUES 1, 'a';
+INSERT INTO hive_catalog.default.sample SELECT id, data from other_kafka_table;
+```
+
+### `INSERT OVERWRITE`
+
+To replace data in the table with the result of a query, use `INSERT OVERWRITE`. Overwrites are atomic operations for Iceberg tables.
+
+Partitions that have rows produced by the SELECT query will be replaced, for example:
+
+```sql
+INSERT OVERWRITE hive_catalog.default.sample VALUES 1, 'a'
+```
+
+Iceberg also support overwriting given partitions by the `select` values:
+
+```sql
+INSERT OVERWRITE hive_catalog.default.sample PARTITION(data='a') SELECT 6;
```

Comment:
The [documentation for overwrites](https://ci.apache.org/projects/flink/flink-docs-release-1.9/api/java/org/apache/flink/table/sinks/PartitionableTableSink.html#getPartitionFieldNames--) doesn't account for hidden partitions.

> When all the partition columns are set a value in PARTITION clause, it is inserting into a static partition. . . . When partial partition columns (prefix part of all partition columns) are set a value in PARTITION clause, it is writing the query result into a dynamic partition.

It would be great to clarify how hidden partitions are handled in that documentation, but it definitely needs to be covered here.

I think this should also specifically say that unpartitioned tables are completely overwritten by `INSERT OVERWRITE`.